### PR TITLE
Change action rule classifiers to a sql where clause

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
Allow SQL instead of classifiers

# What has changed
remove action rule classifiers, new userDefinedWhereClause

# Checklist
Reminder: Make you have run `build_groundzero_ddl.sh` and not manually edited the ground zero DDL.
* [x] Have run automated script, not made manual changes

Reminder: Make sure to version tag this after the PR is merged
* [x] Updated patch number
* [x] Updated version_tag in `ddl_version.sql`
* [x] Updated current_version in `patch_database.py`

https://trello.com/c/TnldEmYZ/1023-address-frame-delta-generate-additional-initial-contact-files-13

### CODE REVIEW

File: patches/2300_make_classifier_a_varchar.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+TRUNCATE actionv2.action_rule CASCADE;
+
+ALTER TABLE actionv2.action_rule
+    DROP COLUMN classifiers;
+
+ALTER TABLE actionv2.action_rule
+    ADD COLUMN IF NOT EXISTS user_defined_where_clause VARCHAR(255) NOT NULL;
```
Comment (AdamHawtin) [inline patches/2300_make_classifier_a_varchar.sql:8]: Still setting 255 chars here

**Comment by AdamHawtin** [comment]
Approved, the PR checklist items need attention though

File: groundzero_ddl/actionv2.sql [hunk 2]
```diff
@@ -9,9 +9,9 @@
     create table action_rule (
        id uuid not null,
         action_type varchar(255),
-        classifiers jsonb not null,
         has_triggered boolean,
         trigger_date_time timestamp with time zone,
+        user_defined_where_clause varchar(100000) not null,
```
Comment (NickGrantONS) [inline groundzero_ddl/actionv2.sql:14]: not user defined; a classifier

Comment (NickGrantONS) [reply groundzero_ddl/actionv2.sql:14]: would be better as `TEXT` type

File: patches/2300_make_classifier_a_varchar.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+TRUNCATE actionv2.action_rule CASCADE;
+
+ALTER TABLE actionv2.action_rule
+    DROP COLUMN classifiers;
+
+ALTER TABLE actionv2.action_rule
+    ADD COLUMN IF NOT EXISTS user_defined_where_clause VARCHAR(100000) NOT NULL;
```
Comment (NickGrantONS) [inline patches/2300_make_classifier_a_varchar.sql:8]: see other comment

File: patches/2200_add_created_date_time_columns.sql [hunk 4]
```diff
@@ -1,7 +1,4 @@
-ALTER TABLE actionv2.cases ADD COLUMN IF NOT EXISTS created_date_time timestamp with time zone;
-
-ALTER TABLE actionv2.cases ADD COLUMN IF NOT EXISTS last_updated timestamp with time zone;
-
-UPDATE actionv2.cases AC SET created_date_time = CC.created_date_time FROM casev2.cases CC WHERE CC.case_id = AC.case_id AND AC.created_date_time is null;
-
-UPDATE actionv2.cases AC SET last_updated = CC.last_updated FROM casev2.cases CC WHERE CC.case_id = AC.case_id AND AC.last_updated is null;
\ No newline at end of file
+ALTER TABLE actionv2.cases ADD COLUMN IF NOT EXISTS created_date_time timestamp with time zone;  commit;
```
Comment (NickGrantONS) [inline patches/2200_add_created_date_time_columns.sql:1]: you can't really change old patches once they're deployed... or rather you shouldn't - it's confusing and might lead to unpredictable behaviour

File: groundzero_ddl/actionv2.sql (hunk 2 above)
Comment (NickGrantONS) [reply groundzero_ddl/actionv2.sql:14]: this type still hasn't been changed

**Pull request merged**